pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -428,7 +428,17 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1alpha1.PipelineRun) er
 		c.Logger.Errorf("Error getting potential next tasks for valid pipelinerun %s: %v", pr.Name, err)
 	}
 
-	rprts := pipelineState.GetNextTasks(candidateTasks)
+	nextRprts := pipelineState.GetNextTasks(candidateTasks)
+	if err := resources.ResolveParams(pipelineState, nextRprts); err != nil {
```

Comment:
We have some existing code to perform variable substitutions in `pkg/reconciler/pipelinerun/resources/apply.go` and I am wondering if we can use some of that to reduce the amount of new machinery we're introducing with `ResolveParams`.

The way that `ApplyParameters` works, for example, is to build up a table of string keys of the format "params.foo" to the value which that param is set to.  So you might end up with a map looking something like this:
```go
map[string]string {
  "params.branch-name": "master",
  "params.repo-url": "https://github.com/tektoncd/pipeline",
}
```

and then ApplyParameters will use the `ApplyReplacements` func to walk through all the tasks in the pipelinespec and perform substitutions on all those task's params.

Would it be possible to reuse some of that code to write an additional func like `ApplyResults` that performs similar work?  It looks like this same approach could map well to the problem we're solving here and would reduce the amount of new code required.  It may also allow us to support the use case I mentioned in another comment of embedding a result's value inside a larger string such as "ci-$(tasks.git-push.results.short-sha)-ephemeral-env".